Source/EasyNetQ/Consumer/InternalConsumer.cs

```diff
@@ -153,9 +179,15 @@ public InternalConsumerStatus StartConsuming(bool firstStart)
             if (queue.IsExclusive && !firstStart)
                 continue;
 
-            if (consumers.ContainsKey(queue.Name))
+            if (cancelledConsumers.Contains(queue.Name))
```

Comment:
Even without this most of the time a cancelled consumer is not able to start again because there is no queue.